src/analysis/convex/cone.lean

```diff
@@ -707,4 +737,87 @@ begin
   exact is_closed_Ici.preimage (by continuity),
 end
 
+lemma pointed_of_nonempty_closed_convex_cone
+  {K : convex_cone ℝ H} (ne : (K : set H).nonempty) (hc : is_closed (K : set H)) : K.pointed :=
+begin
+  obtain ⟨x, hx⟩ := ne,
+  let f : ℝ → H := (• x),
+
+  -- f (0, ∞) is a subset of K
+  have fI : f '' set.Ioi 0 ⊆ (K : set H),
+  { rintro _ ⟨_, h, rfl⟩,
+    exact K.smul_mem (set.mem_Ioi.1 h) hx },
+
+  -- closure of f (0, ∞) is a subset of K
+  have clf : closure (f '' set.Ioi 0) ⊆ (K : set H) := hc.closure_subset_iff.2 fI,
+
+  -- f is continuous at 0 from the right
+  have fc : continuous_within_at f (set.Ioi (0 : ℝ)) 0 :=
+    (continuous_id.smul continuous_const).continuous_within_at,
+
+  -- 0 belongs to the closure of the f (0, ∞)
+  have mem₀ := fc.mem_closure_image (by rw [closure_Ioi (0 : ℝ), set.mem_Ici]),
+  have f₀ : f 0 = 0 := zero_smul ℝ x,
+  rw f₀ at mem₀,
+
+  -- as 0 ∈ closure f (0, ∞) and closure f (0, ∞) ⊆ K, 0 ∈ K.
+  rw [convex_cone.pointed, ← set_like.mem_coe],
+  exact set.mem_of_subset_of_mem clf mem₀,
+end
+
+section complete_space
+variables [complete_space H]
+
+/-- This is a stronger version of the Hahn-Banach separation theorem for closed convex cones. This
+is also the geometric interpretation of Farkas' lemma. -/
+theorem hyperplane_separation_point_nonempty_closed_convex_cone {K : convex_cone ℝ H}
+  (ne : (K : set H).nonempty) (hc : is_closed (K : set H)) {b : H} (disj : b ∉ K) :
+  ∃ (y : H), (∀ x : H, x ∈ K → 0 ≤ ⟪x, y⟫_ℝ) ∧ ⟪y, b⟫_ℝ < 0 :=
```

Comment:
I am not sure whether this is a good name, but I have no good suggestions. I was thinking something along the lines of `exists_separating_of_nonempty_closed_convex_cone`

Reply:
I was using this name as it's the name of a standard theorem: https://en.wikipedia.org/wiki/Hyperplane_separation_theorem

Reply:
A closely related theorem in the library is: [geometric_hahn_banach_point_closed](https://leanprover-community.github.io/mathlib_docs/analysis/normed_space/hahn_banach/separation.html#geometric_hahn_banach_point_closed)

Reply:
ah ok, then it is probably fine